core/modules/installer/installer.pages.inc

```diff
@@ -650,7 +650,18 @@ function installer_browser_installation_enable_form($form, &$form_state, $projec
   foreach ($projects as $project) {
     if ($project['type'] == 'module' && isset($modules[$project['name']])) {
       foreach ($modules as $module) {
-        if (isset($module->info['project']) && isset($modules[$project['name']]->info['project']) && $modules[$project['name']]->info['project'] == $module->info['project'] && $modules[$project['name']]->name != $module->name && !$module->info['hidden']) {
+        // Ignore stand-alone and hidden modules.
+        if (!isset($module->info['project']) || !isset($modules[$project['name']]->info['project']) || !empty($module->info['hidden'])) {
+          continue;
+        }
+        // The type key is required, skip over the violating module, but log.
+        if (!isset($module->info['type'])) {
+          $message = t('Project %project is missing the required type key in its info file.');
+          watchdog('installer', $message, array('%project' => $module->info['name']), WATCHDOG_WARNING);
```

Comment:
The word "project" is reserved for the "main" module + all submodules it includes. Since we'll be catching individual (sub)modules here, we should use the word "module" instead:

```suggestion
          $message = t('The %module module is missing the required "type" key in its .info file.');
          watchdog('installer', $message, array('%module' => $module->info['name']), WATCHDOG_WARNING);
```

Reply:
...also, how about adding this to the logged message?:

`See: https://api.backdropcms.org/modules#info-files`

Reply:
> ...also, how about adding this to the logged message?

Might make sense... On the other hand the person that sees this message is very likely not the maintainer/developer, but rather an admin / site builder. Could this be confusing for them?